### ISSUE TITLE
bug(dialog): onStateChange and onOpen and onClose fire out of order

### ISSUE THREAD

**Issue by mgr34** [jamesmfriedman/rmwc#569]
I am noting on 6.0.0-alpha15 that `onStateChange` does not sync as expected with `onOpen` and `onClose`

take for example (https://codesandbox.io/s/rmwc-sandbox-8ejxb)

```js
function Example() {
  const [open, setOpen] = React.useState(false);
  return (
    <React.Fragment>
      <Dialog
        open={open}
        onOpen={() => console.log("OPEN")}
        onClose={evt => {
          console.log("CLOSED");
          setOpen(false);
        }}
        onStateChange={state => console.log({ state })}
      >
        <DialogTitle>Dialog Title</DialogTitle>
        <DialogContent>This is a standard dialog.</DialogContent>
        <DialogActions>
          <DialogButton action="close">Cancel</DialogButton>
          <DialogButton action="accept" isDefaultAction>
            Sweet!
          </DialogButton>
        </DialogActions>
      </Dialog>

      <Button raised onClick={() => setOpen(true)}>
        Open standard Dialog
      </Button>
    </React.Fragment>
  );
}
```

Console output is as follows

```
OPEN 
{state: "opening"}
{state: "opened"}
CLOSED
{state: "closing"}
{state: "closed"}
```

I would expect  OPEN to come after opening and CLOSED  to come after closing. A reasonable work around would be to only use `onStateChange` when needing that sort of control. Except, `onStateChange` does not pass the action. Which seems to be the way the foundation operates.

**Comment by mgr34** [comment]
Ah, this looks like an issue where documentation needs to be updated. 

```typescript
notifyOpening: () => {
          emit('onOpen', {});
          getProps().onStateChange?.('opening');
        },
        notifyOpened: () => {
          emit('onOpened', {});
          getProps().onStateChange?.('opened');
        },
        notifyClosing: (action: string) => {
          emit('onClose', action ? { action } : {});
          getProps().onStateChange?.('closing');
        },
        notifyClosed: (action: string) => {
          emit('onClosed', action ? { action } : {});
          getProps().onStateChange?.('closed');
        },
```

**Comment by mgr34** [comment]
closed with #570
